org.adempiere.base/src/org/compiere/model/MInvoice.java

```diff
@@ -3347,4 +3351,70 @@ public void updateFrom(MOrder order)
 			}
 		}
 	}
+	
+	/**
+	 * Allocate Invoice for Completed Unallocated Pre-payments
+	 */
+	private boolean allocateFromOrder() {
+		// Get Set Of OrderIds from Lines
+		HashSet<Integer> orderIDSet = new HashSet<>();
+
+		if(getC_Order_ID() > 0)
+			orderIDSet.add(getC_Order_ID());
+
+		MInvoiceLine[] lines = getLines();
+		for(MInvoiceLine line : lines) {
+			if(line.getC_OrderLine_ID() <= 0)
+				continue;
+
+			 I_C_OrderLine orderLine = line.getC_OrderLine();
+			 if(orderLine == null)
+				 continue;
+
+			 orderIDSet.add(orderLine.getC_Order_ID());
+		}

```

Comment:
Should be just `if (line.getC_OrderLine_ID() > 0) orderIDset.add(...)`

Reply:
I means should be simplify to:

   for(MInvoiceLine line : lines) {
	if (line.getC_OrderLine_ID() > 0)
	     orderIDSet.add(orderLine.getC_Order_ID());
 }

Reply:
I see, done